File: darts-ruby/spec/darts/bed/single_spec.rb
require 'darts'

describe Bed::Single do
  it 'should allow a valid bed value' do
    expect(Bed::Single.new(5).bed_score).to eq(5)
  end

  it 'should raise exception when created with bed value just below minimum in range' do
    expect {Bed::Single.new(0) }.to raise_error(ArgumentError, 'Throw score must be in range 1..20')
  end

  it 'should raise exception when created with bed value just above maximum in range' do
    expect {Bed::Single.new(21)}.to raise_error(ArgumentError, 'Throw score must be in range 1..20')
  end
end

File: javascript/spec/darts/bed/single_spec.js
const Single = require('../../../app/darts/bed/single');

describe('Single', () => {
    describe('.bedScore', () => {
        it('should allow a valid bed value', () => {
            expect(new Single(5).bedScore()).toEqual(5);
        });

        it('it should raise exception when created with bed score just below minimum in range', () => {
            expect(() => { new Single(0) }).toThrowError(RangeError, 'Bed score must be in range 1..20');
        });

        it('it should raise exception when created with bed score just above maximum in range', () => {
            expect(() => { new Single(21) }).toThrowError(RangeError, 'Bed score must be in range 1..20');
        });
    });
});